docs/devops.md

```diff
@@ -0,0 +1,237 @@
+---
+name: ' DevOps practices'
+menu: Best practices
+route: /devops
+---
+
+# DevOps practices
+
+---
+
+## Dockerfile guidelines
+
+From Docker's [_Best practices for writing Dockerfiles_](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/):
+
+> Docker builds images automatically by reading the instructions from a Dockerfile -- a text file that contains all commands, in order, needed to build a given image. A Dockerfile adheres to a specific format and set of instructions which you can find at Dockerfile reference.
+> 
+> A Docker image consists of read-only layers each of which represents a Dockerfile instruction. The layers are stacked and each one is a delta of the changes from the previous layer.
+
+### TL;DR
+
+*   Use the provided [base images](https://hub.docker.com/u/helsinkitest).
+
+*   Keep the image slim.
+
+*   One service per image.
+
+*   Use multi-stage builds, last stage should be the production build.
+
+*   Containers are run as root by default which is not nice. Drop root with `USER appuser`.
+
+*   Set owner when copying files: `COPY --chown=appuser:appuser`.
+
+*   Define an entrypoint.
+
+
+### Base image
+
+We have production-ready base images for node, python, and node+python available at [Docker Hub](https://hub.docker.com/u/helsinkitest). These images are based on debian and have
+
+*   a non-root user (`appuser`) for running the application,
+
+*   a directory for application code (`/app`),
+
+*   helpers for installing system packages (`apt-install.sh`, `apt-cleanup.sh`),
+
+*   bash as the shell, and
+
+*   `curl`, `git`, and `wait-for-it.sh` [🔗](https://github.com/vishnubob/wait-for-it).
+
```

Comment:
WRT wait-for-it: Should be specify that applications should strive to handle missing database gracefully? Ie. database migrations should wait for the database to appear. I agree that currently wait-for-it or something similar is needed.